sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/src/ConnectionString/ConnectionStringParser.cs

```diff
@@ -0,0 +1,124 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Linq;
+
+namespace OpenTelemetry.Exporter.AzureMonitor.ConnectionString
+{
+    internal static class ConnectionStringParser
+    {
+        /// <summary>
+        /// Parse a connection string that matches the format: "key1=value1;key2=value2;key3=value3".
+        /// This method will encapsulate all exception handling.
+        /// </summary>
+        /// <remarks>
+        /// Official Doc: <a href="https://docs.microsoft.com/azure/azure-monitor/app/sdk-connection-string" />.
+        /// </remarks>
+        public static void GetValues(string connectionString, out string ikey, out string endpoint)
+        {
+            try
+            {
+                if (connectionString == null)
+                {
+                    throw new ArgumentNullException(nameof(connectionString));
+                }
+                else if (connectionString.Length > Constants.ConnectionStringMaxLength)
+                {
+                    throw new ArgumentOutOfRangeException(nameof(connectionString), $"Values greater than {Constants.ConnectionStringMaxLength} characters are not allowed.");
+                }
+
+                var connString = Azure.Core.ConnectionString.Parse(connectionString);
+                ikey = connString.GetInstrumentationKey();
+                endpoint = connString.GetIngestionEndpoint();
```

Comment:
Consider rename this from `endpoint` to `ingestionEndpoint`.

Reply:
done